datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx

```diff
@@ -169,9 +196,20 @@ export default function SchemaTable({
                         },
                     }}
                     expandable={{
+                        expandedRowKeys: [...Array.from(expandedRows)],
                         defaultExpandAllRows: false,
                         expandRowByClick: false,
                         expandIcon: ExpandIcon,
+                        onExpand: (expanded, record) => {
+                            if (expanded) {
+                                setExpandedRows((previousRows) => new Set(previousRows.add(record.fieldPath)));
+                            } else {
+                                setExpandedRows((previousRows) => {
+                                    previousRows.delete(record.fieldPath);
```

Comment:
tricky!

Reply:
ugh i know - updating Sets in state isn't as simple as you'd want it to be